examples/next/pages/ui/components/storage/storage-browser/default-auth/routed/[locations]/[location-detail]/index.page.tsx

```diff
@@ -11,55 +11,55 @@ import '@aws-amplify/ui-react-storage/styles.css';
 
 export default function Page() {
   const [key, setKey] = React.useState(() => crypto.randomUUID());
-  const router = useRouter();
+  const { replace, query, pathname } = useRouter();
 
-  if (!router.query.bucket) return null;
+  if (!query.bucket) return null;
 
-  const { path, ...location } = router.query;
+  const { path, ...location } = query;
+
+  if (typeof location.permissions === 'string') {
```

Comment:
The permissions must be an array, but in `useRouter` does not deserialize the query into arrays automatically. It means if you only have 1 permission like `list`, you have to have logic like this to make the permissions an array. So this example shows a minimalist way to prevent validation error.

Reply:
In case we use this code for documentation let's add a comment explaining this please

Reply:
makes sense. Thanks. +1 to adding a comment in